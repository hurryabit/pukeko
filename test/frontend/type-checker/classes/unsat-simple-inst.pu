data T a = | C a
class C a where
  c : a -> Int
instance cT : (C a) => C (T a) where
  c t =
    match t with
    | C x -> c x
f : T a -> Int
f = c
