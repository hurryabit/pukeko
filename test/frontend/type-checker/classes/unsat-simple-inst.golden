test/frontend/type-checker/classes/unsat-simple-inst.pu:9:1: cannot deduce C a from context
